http/file_server.ts

```diff
@@ -286,8 +286,14 @@ export async function serveFile(
     headers.set("etag", simpleEtag);
 
     // If a `if-node-match` header is present and the value matches the tag return 304
+    // If a `if-modified-since` header is present and the value is bigger than
+    // the access timestamp value return 304
     const ifNoneMatch = req.headers.get("if-none-match");
-    if (ifNoneMatch && ifNoneMatch === simpleEtag) {
+    const ifModifiedSince = req.headers.get("if-modified-since");
+    if (
+      (ifNoneMatch && ifNoneMatch === simpleEtag) || (ifModifiedSince &&
+        fileInfo.mtime.getTime() < (new Date(ifModifiedSince).getTime() + 1000))
```

Comment:
why `+ 1000`?

Reply:
@kt3k because the `If-Modified-Since` has this format: `<day-name>, <day> <month> <year> <hour>:<minute>:<second> GMT` - meaning it doesn't break down to milliseconds. Therefore, the converted value (number of milliseconds* since the Unix Epoch) from `If-Modified-Since` will always be 1-1000 milliseconds lower.

Reply:
Thanks for the explanation. Makes sense.